src/test/ui/const-generics/issues/issue-56445-1.min.stderr

```diff
@@ -6,6 +6,15 @@ LL | struct Bug<'a, const S: &'a str>(PhantomData<&'a ()>);
    |
    = note: for more information, see issue #74052 <https://github.com/rust-lang/rust/issues/74052>
 
-error: aborting due to previous error
+error: `&'static str` is forbidden as the type of a const generic parameter
```

Comment:
Why is this additional error reported?
I'd expect setting the resolution to `LifetimeRes::Error` to silence derived errors like this, like previously.

Reply:
Same question about additional errors in src/test/ui/const-generics/const-arg-in-const-arg.rs

Reply:
I think this is due to the way typeck handles errors.  `rustc_typeck::check_crate` aborts compilation earlier in case a diagnostic has been emitted.  As `resolve_lifetimes` was called by typechecking, the diagnostics emitted by lifetime resolution were triggered a compilation abort. Now that diagnostics are emitted elsewhere, compilation continues normally.

Reply:
Could you then update tests to avoid off-topic errors when possible?
E.g. make lifetimes in `fn faz`/`fn baz` early-bound so that `faz::<'a>` doesn't produce an additional error, etc.

Reply:
I did not find a satisfactory way to avoid those errors.
If I understand the other test correctly, the difference between baz and faz is that baz is early-bound and faz is late-bound. I don't think changing faz to be early-bound is the right thing here.

Reply:
Ok.